arangod/Cluster/ServerState.cpp

```diff
@@ -285,6 +285,153 @@ ServerState::RoleEnum ServerState::getRole() {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// @brief try to register with a role
+////////////////////////////////////////////////////////////////////////////////
+bool ServerState::registerWithRole(ServerState::RoleEnum role) {
+  if (!getId().empty()) {
+    LOG(INFO) << "Registering with role and localinfo. Supplied id is being ignored";
+    return false;
+  }
+
+  AgencyComm comm;
+  AgencyCommResult result;
+  // mop: hmm...why is it below target? :S
+  result = comm.getValues("Target/MapLocalToID/" + StringUtils::urlEncode(_localInfo), false);
```

Comment:
Let's move it to /MapLocalToID, only need to adjust the deployment scripts (discover.js).